### PR TITLE
update to last changes on main

### PR DESCRIPTION
**Pull request by devnexen**

### CODE REVIEW

**Comment by SchrodingerZhu** [comment]
ping @mjp41

File: snmalloc-sys/src/lib.rs [hunk 1]
```diff
@@ -57,14 +57,14 @@ extern "C" {
...
+    pub fn realloc(p: *mut c_void, newsize: usize) -> *mut c_void;
 
     /// Free previously allocated memory.
     /// The pointer `p` must have been allocated before (or be null).
-    pub fn sn_free(p: *mut c_void);
+    pub fn free(p: *mut c_void);
 
     /// Return the available bytes in a memory block.
-    pub fn sn_malloc_usable_size(p: *const c_void) -> usize;
+    pub fn sn_rust_usable_size(p: *const c_void) -> usize;
```
Comment (mjp41) [inline snmalloc-sys/src/lib.rs:67]: Do you still want the `sn_` prefix?

**Review by mjp41 — APPROVED** [review]
LGTM.  I am hoping to land the BatchIt PR from @nwf, and then do a release for the main repo.  So it might be worth not updating the crate until that is there as well.

**Comment by SchrodingerZhu** [comment]
The CI testing is experiencing segmentation faults.

File: snmalloc-sys/src/lib.rs (hunk 1 above)
Comment (devnexen) [reply snmalloc-sys/src/lib.rs:67]: Let s say I m not bothered by it. Might give a clearer path of its origin tough.

**Comment by devnexen** [comment]
yes I put it i draft for the time being, I can reproduce locally, I ll be back to it once I have more time.

**Review by SchrodingerZhu — APPROVED** [review]

**Pull request merged**